woocommerce/woocommerce-android
# [REST API] Introduce a new screen to handle site credentials login

Part of: #8110 

### Description
This PR is the first part of #8110, the goal here is to introduce a new screen that handles the site credentials authentication instead of overriding the one from the login library, and the main reason for this is that the `LoginUsernamePasswordFragment` in the login library does a lot more than just a simple authentication, so adding the checks we need to execute before proceeding to the authentication would be challenging.

The introduced screen matches the current one 100%, but hopefully it should be easier to customize, so we can improve the error handling if needed, and it's testable too.

### Testing instructions
Repeat the same steps as #8052, and confirm the same flow is happening.
Test also some error scenarios, and confirm it works as expected.
